src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -129,6 +133,21 @@ public function exists($files)
             if (!file_exists($file)) {
                 return false;
             }
+
+            if (null !== $ofType) {
+                switch ($ofType) {
+                    case self::FILE_TYPE_REGULAR:
+                        if (!is_file($file)) {
+                            return false;
+                        }
+                        break;
```

Comment:
Note that files could be iterable so you would return if the first is true but not the second.
Please add a corresponding testcase

Reply:
Done :)